src/common/base/src/containers/pool.rs

```diff
@@ -75,12 +77,13 @@ where
     Mgr::Item: Clone + Sync + Send + Debug,
     Mgr::Error: Sync + Debug,
 {
-    pub fn new(manager: Mgr, initial_retry_interval: Duration) -> Self {
+    pub fn new(manager: Mgr, initial_retry_interval: Duration, retries: Option<u32>) -> Self {
```

Comment:
I prefer to add a new API called `with_retries(self, retries: u32) -> Self` for `Pool`.

- Users who don't care about retry times can use `Pool::new()` AS-IS.
- Users who want to change the default retries can use `Pool::new().with_retries(123)`.

Reply:
I love your approach more. I'll update the PR later, not accessible to my work machine now. BTW do you know what's the problem with the CI? Is there something wrong with my code, or just some random glitches?

Reply:
> BTW do you know what's the problem with the CI? Is there something wrong with my code, or just some random glitches?

Sorry for my lateness. I'm going out today.

Based on the error message, I think we need to update the test:

```rust
running 13 tests
test pool::test_pool ... FAILED
test progress::test_progress ... ok
test range_key_test::test_range_wrapper_reflexive_property ... ok
test range_map_test::test_range_set ... ok
test runtime::test_runtime ... ok
test pool_retry::test_pool ... ok
test stoppable::test_stop_handle ... ok
test stoppable::test_stop_handle_drop ... ok
test stoppable::test_stoppable ... ok
test string_func::mask_string_test ... ok
test string_func::replace_nth_char_test ... ok
test string_func::test_progress ... ok
test runtime::test_shutdown_long_run_runtime ... ok

failures:

---- pool::test_pool stdout ----
thread 'pool::test_pool' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `3`: seq=2 valid for make', src/common/base/tests/it/pool.rs:37:5

failures:
    pool::test_pool
```